src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -362,12 +401,20 @@ private String generateAssertionEntryPointMethodsFor(final Set<ClassDescription>
       String assertionEntryPointMethodContent = assertionEntryPointMethodTemplate.getContent();
       // resolve class assert (ex: PlayerAssert)
       // in case of inner classes like Movie.PublicCategory, class assert will be MoviePublicCategoryAssert
+      String customAssertionClass = fullyQualifiedAssertClassName(classDescription);
+      if (imports.contains(customAssertionClass)) {
```

Comment:
We need it so we can choose whether to use the simple class name or the FQN. Maybe better would be to pass a map from simple class name to FQN and if the simple class name matches and the FQN matches then we use that otherwise we use the FQN (Java 8 would be awesome here 😄).

The imports are created outside of this. I agree with you the best approach would be to compute the imports here (we already are iterating over the `ClassDescription`, but in order to do that we need to change the return so we can later add them to the entry point.